### PR TITLE
Add interface to MigrationInterceptorProvider

### PR DESCRIPTION
**Pull request by darrenhull**
I have added this interface to allow the functionality to be override and extended. for my project we have inheritance within the documents meaning that the runtime migrations can really mess things up. So this allows me to but a noop serializer in.

### CODE REVIEW

**Comment by SRoddis** [comment]
Hey darrenhull,

nice work. Your Pull Request results in a failing build. Did you run the tests? Please run the test and fix the Problem. Looks like the interface cant be resolved in the test. When you fix the build i cant merge the pull request.

https://ci.appveyor.com/project/SRoddis/mongo-migration/builds/32226216

Cheers, 
Sean

File: Mongo.Migration/Migrations/Locators/MigrationLocator.cs [hunk 1]
```diff
@@ -77,7 +77,8 @@ public DocumentVersion GetLatestVersion(Type type)
         {         
             var migrations = GetMigrations(type);
 
-            return migrations.Max(m => m.Version);
+            return migrations?.Max(m => m.Version)
+                ?? new DocumentVersion(0,0,0);
```
Comment (SRoddis) [inline Mongo.Migration/Migrations/Locators/MigrationLocator.cs]: Use DocumentVersion default for this. Its a struct. 

` return migrations?.Max(m => m.Version) ?? DocumentVersion.Default();`

Comment (SRoddis) [reply Mongo.Migration/Migrations/Locators/MigrationLocator.cs]: I needet to add this line of code to fix another issue. You can remove it from the PR.

**Comment by darrenhull** [comment]
I have removed the migrations?.Max(m => m.Version)
                ?? new DocumentVersion(0,0,0); and fixed the tests. (I think couldnt get netstandard 2.0 to run but 2.2 ran and passed. will check appveyer result.)

**Comment by darrenhull** [comment]
The tests passed but looks like it is missing an Environment variable which resulted in a failure:
No token found in Environment Variable 'COVERALLS_REPO_TOKEN'.
Command exited with code 1

**Comment by SRoddis** [comment]
Yes, i changed the token. That has nothing todo with your commit. I merged your pull request, please update to 3.0.112

Cheers, Sean

**Pull request merged**